lib/python/ctypes/ctypesgencore/parser/preprocessor.py

```diff
@@ -169,8 +169,26 @@ def parse(self, filename):
                               shell=True,
                               universal_newlines=True,
                               stdout=subprocess.PIPE,
-                              stderr=subprocess.PIPE)
-        ppout, pperr = pp.communicate()
+                              stderr=subprocess.PIPE)        
+        try:
+            ppout, pperr = pp.communicate()
+        except ValueError:
+            # Fix for https://trac.osgeo.org/grass/ticket/3883,
+            # handling file(s) encoded with mac_roman
+            if sys.platform == 'darwin':
+                pp = subprocess.Popen(cmd,
+                                      shell=True,
+                                      stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE)
```

Comment:
Perhaps `universal_newlines=False` to be explicit about the unicode. A comment might be nice to.

Reply:
good point